pkg/kubelet/cpumanager/cpu_manager.go

```diff
@@ -58,23 +57,30 @@ type Manager interface {
 	State() state.Reader
 }
 
-func NewManager(p Policy, cr internalapi.RuntimeService, lister podLister, statusProvider status.PodStatusProvider) (Manager, error) {
-	cpuInfoFile, err := ioutil.ReadFile("/proc/cpuinfo")
-	if err != nil {
-		return nil, err
-	}
-	topo, err := discoverTopology(cpuInfoFile)
-	if err != nil {
-		return nil, err
+func NewManager(policyTYpe string, cr internalapi.RuntimeService, kletGetter kletGetter, statusProvider status.PodStatusProvider) (Manager, error) {
+	var newPolicy Policy
+
+	switch policyTYpe {
+	case "noop":
```

Comment:
agree for internals of cpu manager but do you want to use this type also outside ?
`type KubeletConfiguration struct` has multiple variables that are limited in terms of values (`ContainerRuntime`,`CgroupDriver`) and the types are kept as simple types